### PR TITLE
Change PLG Chat default to Sonnet

### PR DESCRIPTION
**Pull request by philipp-spiess**
Now that we have https://github.com/sourcegraph/cody/issues/3413 and https://github.com/sourcegraph/sourcegraph/pull/61396. Let's change the default for PLG users (free and pro) 

## Test plan

- CI

### CODE REVIEW

**Review by umpox — APPROVED** [review]

File: lib/shared/src/models/dotcom.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const DEFAULT_DOT_COM_MODELS = [
         title: 'Claude 3 Sonnet',
         model: 'anthropic/claude-3-sonnet-20240229',
         provider: 'Anthropic',
-        default: false,
+        default: true,
         codyProOnly: true,
```
Comment (abeatrix) [inline lib/shared/src/models/dotcom.ts:48]: Is it still Cody Pro only?

**Review by abeatrix — COMMENTED** [review]
Based on the pr sonnet is no longer Cody pro only, right?

**Comment by philipp-spiess** [comment]
re-requesting review. There were some bugs as to when the default model is loaded which are fixed now (I created a new user account to test this)

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts [hunk 2]
```diff
@@ -740,7 +740,12 @@ export class SimpleChatPanelProvider implements vscode.Disposable, ChatSession {
         if (!authStatus?.isLoggedIn) {
             return
         }
-        const models = ModelProvider.getProviders(ModelUsage.Chat, this.chatModel.modelID)
+        const models = ModelProvider.getProviders(
+            ModelUsage.Chat,
+            authStatus.isDotCom && !authStatus.userCanUpgrade,
+            this.chatModel.modelID
+        )
+        console.log('getModel', this.chatModel.modelID, models)
```
Comment (umpox) [inline vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:748]:
```suggestion
```

**Review by umpox — APPROVED** [review]
LGTM, ty for testing and fixing

**Pull request merged**